src/databricks/labs/remorph/reconcile/recon_capture.py

```diff
@@ -27,13 +37,69 @@ def _write_df_to_delta(df: DataFrame, table_name: str, mode="append"):
         raise WriteToTableException(message) from e
 
 
+def generate_final_reconcile_output(recon_id: str, spark: SparkSession) -> ReconcileOutput:
+    recon_df = spark.sql(
+        f"""
+    SELECT 
+    CASE 
+        WHEN COALESCE(MAIN.SOURCE_TABLE.CATALOG, '') <> '' THEN CONCAT(MAIN.SOURCE_TABLE.CATALOG, '.', MAIN.SOURCE_TABLE.SCHEMA, '.', MAIN.SOURCE_TABLE.TABLE_NAME) 
+        ELSE CONCAT(MAIN.SOURCE_TABLE.SCHEMA, '.', MAIN.SOURCE_TABLE.TABLE_NAME) 
+    END AS SOURCE_TABLE, 
+    CONCAT(MAIN.TARGET_TABLE.CATALOG, '.', MAIN.TARGET_TABLE.SCHEMA, '.', MAIN.TARGET_TABLE.TABLE_NAME) AS TARGET_TABLE, 
+    CASE WHEN lower(MAIN.report_type) in ('all', 'row', 'data') THEN
+    CASE 
+        WHEN METRICS.recon_metrics.row_comparison.missing_in_source = 0 AND METRICS.recon_metrics.row_comparison.missing_in_target = 0 THEN TRUE 
+        ELSE FALSE 
+    END 
+    ELSE NULL END AS ROW, 
+    CASE WHEN lower(MAIN.report_type) in ('all', 'data') THEN
+    CASE 
+        WHEN METRICS.recon_metrics.column_comparison.absolute_mismatch = 0 AND METRICS.recon_metrics.column_comparison.threshold_mismatch = 0 AND METRICS.recon_metrics.column_comparison.mismatch_columns = '' THEN TRUE 
+        ELSE FALSE 
+    END 
+    ELSE NULL END AS COLUMN, 
+    CASE WHEN lower(MAIN.report_type) in ('all', 'schema') THEN
+    CASE 
+        WHEN METRICS.recon_metrics.schema_comparison = true THEN TRUE 
+        ELSE FALSE 
+    END
+    ELSE NULL END AS SCHEMA,
+    METRICS.run_metrics.exception_message AS EXCEPTION_MESSAGE 
+    FROM 
+        {_DB_PREFIX}.{_RECON_TABLE_NAME} MAIN 
+    INNER JOIN 
+        {_DB_PREFIX}.{_RECON_METRICS_TABLE_NAME} METRICS 
+    ON 
+        (MAIN.recon_table_id = METRICS.recon_table_id) 
+    WHERE 
+        MAIN.recon_id = '{recon_id}' 
+    """
+    )
+    table_output = []
+    for row in recon_df.collect():
+        if row.EXCEPTION_MESSAGE is not None and row.EXCEPTION_MESSAGE != "":
+            table_output.append(
+                ReconcileTableOutput(
+                    target_table_name=row.TARGET_TABLE,
+                    source_table_name=row.SOURCE_TABLE,
+                    status=StatusOutput(),
+                    exception_message=row.EXCEPTION_MESSAGE,
+                )
+            )
+        else:
+            table_output.append(
+                ReconcileTableOutput(
+                    target_table_name=row.TARGET_TABLE,
+                    source_table_name=row.SOURCE_TABLE,
+                    status=StatusOutput(row=row.ROW, column=row.COLUMN, schema=row.SCHEMA),
+                    exception_message=row.EXCEPTION_MESSAGE,
+                )
+            )
+
+    return ReconcileOutput(recon_id=recon_id, results=table_output)
```

Comment:
can you log before returning the object so useful for the job output log.

Reply:
Done